### PR TITLE
 (refactor)controller: add a struct to hold application info

### PR DESCRIPTION
**Pull request by amitbhatt818**
Signed-off-by: Amit Bhatt <amitbhatt818@gmail.com>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -87,6 +87,23 @@ type ReconcileChaosEngine struct {
 	client client.Client
 	scheme *runtime.Scheme
 }
+type applicationInfo struct {
+	label     map[string]string
+	namespace string
+	expList   []litmuschaosv1alpha1.ExperimentList
+	lKey      string
+	lValue    string
+}
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Update the variable name accordingly like:
```
	namespace  string
	labelKey      string
	labelValue    string
        appLabel     map[string]string
        expperimentList   []litmuschaosv1alpha1.ExperimentList
```

**Comment by ksatchit** [comment]
Let us rebase and resolve conflicts..

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 2]
```diff
@@ -101,6 +101,25 @@ type ReconcileChaosEngine struct {
...
+type applicationInfo struct {
+	namespace      string
+	label          map[string]string
+	experimentList []litmuschaosv1alpha1.ExperimentList
+}
+
+var labelKey string
+var labelValue string
+
+func (a *applicationInfo) Initialize(instance *litmuschaosv1alpha1.ChaosEngine) *applicationInfo {
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: The function name should be something like: **initializeApplicationInfo**

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 3]
```diff
@@ -101,6 +101,25 @@ type ReconcileChaosEngine struct {
...
+	aLabelKeyValue := strings.Split(instance.Spec.Appinfo.Applabel, "=")
+	labelKey = aLabelKeyValue[0]
+	labelValue = aLabelKeyValue[1]
+	a.label = make(map[string]string)
+	a.label[labelKey] = labelValue
+	a.namespace = instance.Spec.Appinfo.Appns
+	a.experimentList = instance.Spec.Experiments
+	return a
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: Here the variable should be like:
```
appLabel := strings.Split(instance.Spec.Appinfo.Applabel, "=")
	appLabelKey = appLabel[0]
	appLabelValue = appLabel[1]
	appInfo.label = make(map[string]string)
	appInfo.label[labelKey] = labelValue
	appInfo.namespace = instance.Spec.Appinfo.Appns
	appInfo.experimentList = instance.Spec.Experiments
	return appInfo
```

File: pkg/controller/chaosengine/chaosengine_controller.go (hunk 2 above)
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]: func (appInfo *applicationInfo) Initialize(instance *litmuschaosv1alpha1.ChaosEngine) *applicationInfo {

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 4]
```diff
@@ -131,16 +150,11 @@ func (r *ReconcileChaosEngine) Reconcile(request reconcile.Request) (reconcile.R
...
 
-	aLabelKeyValue := strings.Split(instance.Spec.Appinfo.Applabel, "=")
-	lKey := aLabelKeyValue[0]
-	lValue := aLabelKeyValue[1]
-	aLabel := make(map[string]string)
-	aLabel[lKey] = lValue
-	aNamespace := instance.Spec.Appinfo.Appns
-	aExpList := instance.Spec.Experiments
+	a := &applicationInfo{}
+	a = a.Initialize(instance)
```
Comment (chandankumar4) [inline pkg/controller/chaosengine/chaosengine_controller.go]:
```
	appInfo := &applicationInfo{}
	appInfo = appInfo.Initialize(instance)
```

**Comment by ksatchit** [comment]
Thanks @chandankumar4 - still getting used to the right naming conventions. oversight.

**Pull request merged**